File: code/test_suites/cpp/FunctionBuildTest/tests/test_matplotlib_plot.cpp
#include <gtest/gtest.h>
#include "../src/function_matplotlib_plot.cpp"

class TestMatplotlibPlot : public ::testing::Test {
};

TEST_F(TestMatplotlibPlot, TestValidInput) {
    std::string result = BarPlotGenerator::create_bar_plot("A,B,C", "10,20,30");
    EXPECT_EQ(result, "Plot saved as 'bar_plot.png'.");
}

TEST_F(TestMatplotlibPlot, TestEmptyInput) {
    std::string result = BarPlotGenerator::create_bar_plot("", "");
    EXPECT_EQ(result, "Error: Categories and values cannot be empty.");
}

TEST_F(TestMatplotlibPlot, TestMismatchedLengths) {
    std::string result = BarPlotGenerator::create_bar_plot("A,B", "10");
    EXPECT_EQ(result, "Error: Categories and values must have the same length.");
}

TEST_F(TestMatplotlibPlot, TestNonStringCategories) {
    std::string result = BarPlotGenerator::create_bar_plot("1,2,3", "10,20,30");
    EXPECT_EQ(result, "Plot saved as 'bar_plot.png'.");
}

TEST_F(TestMatplotlibPlot, TestNonNumericValues) {
    std::string result = BarPlotGenerator::create_bar_plot("A,B,C", "10,'20',30");
    EXPECT_EQ(result, "Error: All values must be numbers.");
}

int main(int argc, char **argv) {
    testing::InitGoogleTest(&argc, argv);
    return RUN_ALL_TESTS();
}

File: data/java/tests/test_matplotlib_plot.java
package com.example;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class test_matplotlib_plot {

    @Test
    void test_valid_input() {
        function_matplotlib_plot plotGenerator = new function_matplotlib_plot();
        String result = plotGenerator.create_bar_plot("A,B,C", "10,20,30");
        assertEquals("Plot saved as 'bar_plot.png'.", result);
    }

    @Test
    void test_empty_input() {
        function_matplotlib_plot plotGenerator = new function_matplotlib_plot();
        String result = plotGenerator.create_bar_plot("", "");
        assertEquals("Error: Categories and values cannot be empty.", result);
    }

    @Test
    void test_mismatched_lengths() {
        function_matplotlib_plot plotGenerator = new function_matplotlib_plot();
        String result = plotGenerator.create_bar_plot("A,B", "10");
        assertEquals("Error: Categories and values must have the same length.", result);
    }

    @Test
    void test_non_string_categories() {
        function_matplotlib_plot plotGenerator = new function_matplotlib_plot();
        String result = plotGenerator.create_bar_plot("1,2,3", "10,20,30");
        assertEquals("Plot saved as 'bar_plot.png'.", result);
    }

    @Test
    void test_non_numeric_values() {
        function_matplotlib_plot plotGenerator = new function_matplotlib_plot();
        String result = plotGenerator.create_bar_plot("A,B,C", "10,'20',30");
        assertEquals("Error: All values must be numbers.", result);
    }
}